package-res/resources/web/pentaho/data/filter/AbstractPropertyFilter.js

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./AbstractFilter",
+  "./_toSpec"
+], function(AbstractFilter, _toSpec) {
+  "use strict";
+
+  /**
+   * @name AbstractPropertyFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @extends pentaho.data.filter.AbstractFilter
+   * @abstract
+   * @amd pentaho/data/filter/AbstractPropertyFilter
+   *
+   * @classdesc A base class for filters that represent a set of elements which
+   * contain a given property
+   * and whose value matches against a reference value.
```

Comment:
No, "elements" is  supposed to be the subject here. All elements have a certain property with a certain value. We want to select those that match a specific value.